File: CPP/Komputer.cpp
#pragma once
#include <iostream>
#include <string>
#include <vector>
#include "Cpu.cpp"
#include "Harddrive.cpp"
#include "Ram.cpp"
#include "Monitor.cpp"


using namespace std;


class Komputer
{
private:
    string nama;
    Cpu cpu;
    vector<Ram> ramList;
    Harddrive harddrive;
    vector<Monitor> monitorList;
public:
    Komputer()
    {


    }


    Komputer(string nama, Cpu cpu, vector<Ram> ramList, Harddrive harddrive, vector<Monitor> monitorList)
    {
        this->nama = nama;
        this->cpu = cpu;
        this->ramList = ramList;
        this->harddrive = harddrive;
        this->monitorList = monitorList;
    }


    void setNama(string nama)
    {
        this->nama = nama;
    }

    void setCpu(Cpu cpu)
    {
        this->cpu = cpu;
    }


    void setRam(vector<Ram> ramList)
    {
        this->ramList = ramList;
    }    
    
    void setHarddrive(Harddrive harddrive)
    {
        this->harddrive = harddrive;
    }
    
    void setMonitor(vector<Monitor> monitorList)
    {
        this->monitorList = monitorList;
    }    
    
    void addRam(Ram ram)
    {
        this->ramList.push_back(ram);
    }
    
    void addMonitor(Monitor monitor)
    {
        this->monitorList.push_back(monitor);
    }

    string getNama()
    {
        return this->nama;
    }


    Cpu getCpu()
    {
        return this->cpu;
    }


    vector<Ram> getRamList() // return vector
    {
        return this->ramList;
    }
    
    
    Harddrive getHarddrive()
    {
        return this->harddrive;
    }
    
    vector<Monitor> getMonitorList() // return vector
    {
        return this->monitorList;
    }

    ~Komputer()
    {


    }
};

File: Python/Komputer.py
from Cpu import Cpu
from Harddrive import Harddrive
from Ram import Ram
from Monitor import Monitor

class Komputer:
    def __init__(self, nama="", cpu = Cpu, ram_list= [], harddrive= Harddrive, monitor_list= []):
        self.nama = nama
        self.cpu = cpu
        self.ram_list = ram_list
        self.harddrive = harddrive
        self.monitor_list = monitor_list

    def set_nama(self, nama):
        self.nama = nama

    def set_cpu(self, cpu):
        self.cpu = cpu

    def set_ram_list(self, ram_list):
        self.ram_list = ram_list

    def set_harddrive(self, harddrive):
        self.harddrive = harddrive

    def set_monitor_list(self, monitor_list):
        self.monitor_list = monitor_list

    def add_ram(self, ram):
        self.ram_list.append(ram)

    def add_monitor(self, monitor):
        self.monitor_list.append(monitor)

    def get_nama(self):
        return self.nama

    def get_cpu(self):
        return self.cpu

    def get_ram_list(self):
        return self.ram_list

    def get_harddrive(self):
        return self.harddrive

    def get_monitor_list(self):
        return self.monitor_list
